core/cloudflow-build-support/src/main/scala/com/lightbend/cloudflow/buildtool/Scaffold.scala

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (C) 2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package com.lightbend.cloudflow.buildtool
+
+import java.nio.file._
+import java.io._
+import com.typesafe.config.{ Config, ConfigFactory }
+import cloudflow.blueprint.deployment._
+import cloudflow.blueprint.deployment.ApplicationDescriptorJsonFormat._
+import spray.json._
+
+import scala.util._
+
+object Scaffold {
+
+  def getDescriptorsOrFail[T](descriptors: Iterable[(T, Try[RuntimeDescriptor])])(
+      error: String => Unit): Iterable[(T, RuntimeDescriptor)] = {
+    descriptors
+      .collect {
+        case (_, Failure(ex)) =>
+          error(s"Determining runtime descriptors failed: ${ex.getMessage}")
+          ex
+      }
+      .foreach { ex =>
+        throw ex
+      }
+    descriptors.collect { case (pid, Success(runtimeDescriptor)) => (pid, runtimeDescriptor) }
+  }
+
+  def scaffoldRuntime(
+      projectId: String,
+      descriptor: ApplicationDescriptor,
+      localConfig: LocalConfig,
+      targetDir: Path,
+      configDir: Path,
+      log4jConfig: Try[Path],
+      write: (File, String) => Unit): Try[RuntimeDescriptor] = {
+    for {
+      appDescriptor <- prepareApplicationDescriptor(descriptor, localConfig.content, targetDir)
+      outputFile <- createOutputFile(targetDir, projectId)
+      logFile <- log4jConfig
+      appDescriptorFile <- prepareApplicationFile(appDescriptor, write)
+    } yield {
+      RuntimeDescriptor(appDescriptor.appId, appDescriptor, appDescriptorFile, outputFile, logFile, localConfig.path)
+    }
+  }
+
+  def streamletFilterByClass(
+      appDescriptor: ApplicationDescriptor,
+      streamletClasses: Set[String]): ApplicationDescriptor = {
+    val streamletInstances =
+      appDescriptor.streamlets.filter(streamlet => streamletClasses(streamlet.descriptor.className))
+    val deploymentDescriptor =
+      appDescriptor.deployments.filter(streamletDeployment => streamletClasses(streamletDeployment.className))
+    appDescriptor.copy(streamlets = streamletInstances, deployments = deploymentDescriptor)
+  }
+
+  def prepareApplicationFile(applicationDescriptor: ApplicationDescriptor, write: (File, String) => Unit): Try[Path] =
+    Try {
+      val localApplicationFile = Files.createTempFile("local-runner", ".json")
+      val contents = applicationDescriptor.toJson.prettyPrint
+      write(localApplicationFile.toFile, contents)
+      localApplicationFile
+    }
+
+  def createOutputFile(workDir: Path, projectId: String): Try[File] = Try {
+    val localFile = workDir.resolve(projectId + "-local.log").toFile
+    if (!localFile.exists()) {
+      Files.createFile(workDir.resolve(projectId + "-local.log")).toFile
+    }
+    localFile
+  }
+
+  def createDirs(prefix: String): (Path, Path) = {
```

Comment:
directory creation can fail. Should we do `Try[(Path, Path)]` ?

Reply:
This looks really aggressive defensive programming 😛 
If directory creation fails I think is correct to send to the user a stack-trace with the original exception (that is what happens here)